esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -547,18 +615,37 @@ def _regrid_rectilinear_to_unstructured__prepare(
     """
     grid_x, grid_y = get_xy_dim_coords(src_grid_cube)
     mesh = target_mesh_cube.mesh
-    # TODO: Improve the checking of mesh validity. Check the mesh location and
-    #  raise appropriate error messages.
+    location = target_mesh_cube.location
+    if mesh is None:
+        raise ValueError("The given cube is not defined on a mesh.")
+    if method == "conservative":
+        if location != "face":
+            raise ValueError(
+                f"Conservative regridding requires a target cube located on "
+                f"the face of a cube, target cube had the {location} location."
+            )
+        center = False
+    elif method == "bilinear":
+        if location not in ["face", "node"]:
+            raise ValueError(
+                f"Bilinear regridding requires a target cube with a node "
+                f"or face location, target cube had the {location} location."
+            )
+        center = True
+    else:
+        raise ValueError("method must be either bilinear or conservative.")
```

Comment:
Use a `NotImplementedError`, referencing the unsupported argument that was provided.